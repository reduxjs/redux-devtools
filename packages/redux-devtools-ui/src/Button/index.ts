export { default } from './Button.js';
