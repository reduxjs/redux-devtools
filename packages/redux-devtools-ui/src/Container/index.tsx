import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';
import { useTheme, ThemeData } from '../utils/theme';
import { MainContainerWrapper, ContainerWrapper } from './styles';
import { Theme } from '../themes/default';

interface ContainerFromThemeDataProps {
  children?: React.ReactNode;
  themeData: ThemeData;
  className?: string;
}

const ContainerFromThemeData: React.FunctionComponent<
  ContainerFromThemeDataProps
> = ({ themeData, className, children }) => {
  const theme = useTheme(themeData);
  return (
    <ThemeProvider theme={theme}>
      <MainContainerWrapper className={className}>
        {children}
      </MainContainerWrapper>
    </ThemeProvider>
  );
};

interface Props {
  children?: React.ReactNode;
  themeData?: ThemeData;
  theme?: Theme;
  className?: string;
}

const Container: React.FunctionComponent<Props> = ({
  themeData,
  className,
  theme,
  children,
}) => {
  if (!themeData) {
    return (
      <ContainerWrapper className={className} theme={theme}>
        {children}
      </ContainerWrapper>
    );
  }

  return (
    <ContainerFromThemeData themeData={themeData} className={className}>
      {children}
    </ContainerFromThemeData>
  );
};

Container.propTypes = {
  children: PropTypes.node,
  themeData: PropTypes.any,
  theme: PropTypes.any,
  className: PropTypes.string,
};

export default Container;
