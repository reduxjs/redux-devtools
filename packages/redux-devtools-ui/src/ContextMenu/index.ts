export { default } from './ContextMenu.js';
