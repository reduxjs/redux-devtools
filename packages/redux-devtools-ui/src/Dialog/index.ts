export { default } from './Dialog.js';
