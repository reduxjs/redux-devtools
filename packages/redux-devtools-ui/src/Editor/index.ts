export { default } from './Editor.js';
