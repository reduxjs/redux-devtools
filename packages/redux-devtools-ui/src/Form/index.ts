export { default } from './Form.js';
