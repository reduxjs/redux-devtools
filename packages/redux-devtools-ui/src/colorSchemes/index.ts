export { default } from './default';
export { default as github } from './github';
export { default as atomOneDark } from './atom-one-dark';
export { default as dracula } from './dracula';
export { default as iRBlack } from './ir-black';
export { default as macintosh } from './macintosh';
export { default as materia } from './materia';
export { default as oceanicNext } from './oceanic-next';
export { default as phD } from './phd';
export { default as pico } from './pico';
export { default as solarFlare } from './solar-flare';
export { default as spacemacs } from './spacemacs';
export { default as unikitty } from './unikitty';
export { default as woodland } from './woodland';
