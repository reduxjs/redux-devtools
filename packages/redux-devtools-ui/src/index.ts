export { default as Container } from './Container';
export { default as Button } from './Button';
export { default as ContextMenu } from './ContextMenu';
export { default as Dialog } from './Dialog';
export { default as Editor } from './Editor';
export { default as Form } from './Form';
export { default as Select } from './Select';
export { default as Slider } from './Slider';
export { default as Tabs, Tab } from './Tabs';
export { default as SegmentedControl } from './SegmentedControl';
export { default as Notification } from './Notification';
export * from './Toolbar';

import color from './utils/color';
export const effects = { color };
export { default as createStyledComponent } from './utils/createStyledComponent';
export { Theme, ThemeFromProvider, Scheme } from './utils/theme';
