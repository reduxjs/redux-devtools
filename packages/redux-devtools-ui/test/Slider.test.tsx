import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { Slider } from '../src';

describe('Slider', function () {
  it('renders correctly', () => {
    const { container } = render(
      <Slider
        onChange={() => {
          // noop
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with props', () => {
    const { container } = render(
      <Slider
        label="Hi"
        min={1}
        max={10}
        value={5}
        disabled
        onChange={() => {
          // noop
        }}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should handle the change event', () => {
    const onChange = jest.fn();
    render(<Slider value={1} onChange={onChange} />);

    fireEvent.change(screen.getByRole('slider'), { target: { value: '2' } });
    expect(onChange).toHaveBeenCalled();
    expect(onChange).toHaveBeenCalledWith(2);
  });
});
