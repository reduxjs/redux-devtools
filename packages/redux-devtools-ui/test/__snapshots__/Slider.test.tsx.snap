// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider renders correctly 1`] = `
<div
  class="sc-gsFSXq kWmynN"
>
  <input
    max="100"
    min="0"
    type="range"
    value="0"
  />
</div>
`;

exports[`Slider renders with props 1`] = `
<div
  class="sc-gsFSXq mpSAc"
  disabled=""
>
  <label>
    Hi
     
  </label>
  <input
    disabled=""
    max="10"
    min="1"
    type="range"
    value="5"
  />
</div>
`;
